pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -86,6 +87,48 @@ func (c *Controller) generatePodUID(p *v1.Pod) string {
 	return c.clusterID.String() + "//" + "Pod/" + p.Namespace + "/" + p.Name
 }
 
+// servicesByNamespacedHostname returns the service with the given namespace and hostname. If
+// the provided namespace is a wildcard ("*"), all services with the hostname are returned.
+func (a *AmbientIndex) servicesByNamespacedHostname(namespace, hostname string) []*model.ServiceInfo {
+	out := make([]*model.ServiceInfo, 0)
+	if services := a.serviceByHostname[hostname]; services != nil {
+		if namespace == "*" {
+			// The query was for all namespaces. Return all services with the hostname.
+			out = append(out, services...)
```

Comment:
nit: return return this directly?

Reply:
I suppose that makes it possible to be modified (probably accidentally) later though. so LGTM